core/templates/dev/head/domain/collection/EditableCollectionBackendApiService.js

```diff
@@ -28,20 +28,22 @@
 // Discuss and decide whether this is a good approach and then remove this TODO
 // after deciding and acting upon the decision (which would mean implementing
 // it if it's agreed upon).
-oppia.factory('WritableCollectionBackendApiService', [
+oppia.factory('EditableCollectionBackendApiService', [
     '$http', '$q', 'COLLECTION_DATA_URL_TEMPLATE',
-    'WRITABLE_COLLECTION_DATA_URL_TEMPLATE', 'UrlInterpolationService',
-    'CollectionBackendApiService',
+    'COLLECTION_EDITABLE_DATA_URL_TEMPLATE', 'UrlInterpolationService',
+    'ReadOnlyCollectionBackendApiService',
     function($http, $q, COLLECTION_DATA_URL_TEMPLATE,
-      WRITABLE_COLLECTION_DATA_URL_TEMPLATE, UrlInterpolationService,
-      CollectionBackendApiService) {
-      var _fetchWritableCollection = function(
+      COLLECTION_EDITABLE_DATA_URL_TEMPLATE, UrlInterpolationService,
+      ReadOnlyCollectionBackendApiService) {
+      var _fetchCollection = function(
           collectionId, successCallback, errorCallback) {
         var collectionDataUrl = UrlInterpolationService.interpolateUrl(
           COLLECTION_DATA_URL_TEMPLATE, {
             collection_id: collectionId
           });
-        collectionDataUrl += '?allow_invalid_explorations=true';
+
+        // Does not work with following line
+        // collectionDataUrl += '?allow_invalid_explorations=true';
```

Comment:
This should implicitly be true for the ``EditableCollectionDataHandler``. Ie, all collections provided by it should always support providing invalid explorations, and the read-only handler should never provide invalid explorations. This arg can be removed from the read-only handler and the editable handler should be updated appropriately.